lib/stats/stat.js

```diff
@@ -0,0 +1,113 @@
+import assert from 'assert';
+import base from 'taskcluster-base';
+
+import * as series from './series';
+
+export default class Stat {
+  constructor(config) {
+    assert(config.influx, 'Must supply an influx configuration');
+    assert(config.workerId, 'Worker ID is required');
+    assert(config.workerType, 'Worker type is required');
+    assert(config.workerGroup, 'Worker group is required');
+    assert(config.workerNodeType, 'Worker instance type is required');
+    assert(config.provisionerId, 'Provisioner ID is required');
+
+    this.influx = new base.stats.Influx(config.influx);
+    this.baseWorkerStats = {
+      workerId: config.workerId,
+      workerGroup: config.workerGroup,
+      workerType: config.workerType,
+      instanceType: config.workerNodeType,
+      provisionerId: config.provisionerId,
+      capacity: config.capacity
+
+    };
+    this.reporters = {};
+  }
+
+  /**
+   * Flushes data and closes influxdb connection.
+   */
+  close() {
+    this.influx.close();
+  }
+
+  /**
+   * Create a reporter for the series.
+   *
+   * @param {String} seriesName - name of the series
+   *
+   * @returns {Series} series object with influx reporter
+   */
+  createReporter(seriesName) {
+    if (!(seriesName in series)) {
+      throw new Error(
+        `Cannot create influx reporter for ${seriesName}.  No ` +
+        `reporter defined`
+      );
+    }
+
+    return series[seriesName].reporter(this.influx);
+  }
+
+  /**
+   * Return a reporter for a given series.
+   *
+   * @param {String} seriesName - name of the series
+   *
+   * @returns {Series} series reporter
+   */
+  getReporter(seriesName) {
```

Comment:
This is total overkill... In the constructor just do:
```js
_.forIn(series, (serie, name) => {
  this[report + name] = series.reporter(this.influx, this.baseWorkerStats);
});
```

Then you'll have all the series initialized initially, and if there is any bugs they'll throw early.
Also the overhead to calling `.reporter()` is very small look at it... It just keeps a reference to `baseWorkerStats` and returns a closure... No real memory allocation of significance. Only a quick few sanity checks.